packages/components/src/components/Address/Address.tsx

```diff
@@ -6,11 +6,27 @@ export interface AddressProps {
    * The address to display
    */
   value: string;
+  /**
+   * boolean to shorten the address if addresses does not resolve to an ENS name
+   */
+  shortened: boolean;
 }
 
 /**
  * A component to display an address
  */
-export const Address: React.FC<AddressProps> = ({ value }) => {
-  return <Input value={value} />;
+export const Address: React.FC<AddressProps> = ({ value, shortened }) => {
+  if (shortened) {
+    if (value.includes('.eth')) {
+      return <Input value={value} />;
+    } else if (value === '' || value === 'Not connected') {
```

Comment:
If I understand correctly you are accounting for an empty instance here.

Perhaps it would be easier to drop this `if statement` in favour of a `placeholder` prop on the `<Input />` instead?

https://chakra-ui.com/docs/form/input#usage

Reply:
I added a check for 'Not connected' because if that value gets passed into the prop it displays the shortened version of it.

C:\Users\power\Downloads\2021-12-12 16_34_14-Components _ Address - Default ⋅ Storybook - Brave.png

If we went with placeholder prop, would that be in addition to passing the value? so `value={value} placeholder={value}`?